diff --git a/mod-aoeloot/src/aoeloot_loader.cpp b/mod-aoeloot/src/aoeloot_loader.cpp
index 52e11a0..e74cc01 100644
--- a/mod-aoeloot/src/aoeloot_loader.cpp
+++ b/mod-aoeloot/src/aoeloot_loader.cpp
@@ -21,7 +21,7 @@ void AddSC_aoeloot();
 
 // The name of this function should match:
 // void Add${NameOfDirectory}Scripts()
-void Addmod_aoelootScripts()
+void Addmod_aoe_lootScripts()
 {
     AddSC_aoeloot();
 }
